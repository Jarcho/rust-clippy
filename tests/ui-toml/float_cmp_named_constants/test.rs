//@no-rustfix

#![deny(clippy::float_cmp)]

fn main() {
    {
        const C: f64 = 1.0;
        fn f(x: f64) {
            let _ = x == C;
        }
    }
    {
        const fn f(x: f64) -> f64 {
            todo!()
        }
        let _ = f(1.0) == f(2.0);
    }
}
