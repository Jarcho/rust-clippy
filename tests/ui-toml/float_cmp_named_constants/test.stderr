error: strict comparison of `f32` or `f64`
  --> tests/ui-toml/float_cmp_named_constants/test.rs:9:21
   |
LL |             let _ = x == C;
   |                     ^^^^^^ help: consider comparing them within some margin of error: `(x - C).abs() < error_margin`
   |
note: the lint level is defined here
  --> tests/ui-toml/float_cmp_named_constants/test.rs:3:9
   |
LL | #![deny(clippy::float_cmp)]
   |         ^^^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

